crates/sui-rpc-api/proto/sui/rpc/v2alpha/checkpoint_summary.proto

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+package sui.rpc.v2alpha;
+
+import "google/protobuf/timestamp.proto";
+import "sui/rpc/v2alpha/bcs.proto";
+import "sui/rpc/v2alpha/gas_cost_summary.proto";
+import "sui/rpc/v2alpha/signature.proto";
+
+// A header for a checkpoint on the Sui blockchain.
+//
+// On the Sui network, checkpoints define the history of the blockchain. They are quite similar to
+// the concept of blocks used by other blockchains like Bitcoin or Ethereum. The Sui blockchain,
+// however, forms checkpoints after transaction execution has already happened to provide a
+// certified history of the chain, instead of being formed before execution.
+//
+// Checkpoints commit to a variety of state, including but not limited to:
+// - The hash of the previous checkpoint.
+// - The set of transaction digests, their corresponding effects digests, as well as the set of
+//   user signatures that authorized its execution.
+// - The objects produced by a transaction.
+// - The set of live objects that make up the current state of the chain.
+// - On epoch transitions, the next validator committee.
+//
+// `CheckpointSummary`s themselves don't directly include all of the previous information but they
+// are the top-level type by which all the information is committed to transitively via cryptographic
+// hashes included in the summary. `CheckpointSummary`s are signed and certified by a quorum of
+// the validator committee in a given epoch to allow verification of the chain's state.
+message CheckpointSummary {
+  // This CheckpointSummary serialized as BCS.
+  optional Bcs bcs = 1;
+
+  // The digest of this CheckpointSummary.
+  optional string digest = 2;
+
+  // Epoch that this checkpoint belongs to.
+  optional uint64 epoch = 3;
+
+  // The height of this checkpoint.
+  optional uint64 sequence_number = 4;
+
+  // Total number of transactions committed since genesis, including those in this
+  // checkpoint.
+  optional uint64 total_network_transactions = 5;
```

Comment:
nit: isn't this called `network_total_transactions` in the core types? It'd be best to keep consistent with that to avoid confusing people.

Reply:
There are a number of names that do differ between the sdk and core mostly for clarity reasons. See GenericSignature vs UserSignature as another example that I have strong opinions on :)

Reply:
To be honest, I think it could just be called `total_transactions` -- I just worry that `network_total_transactions` and `total_network_transactions` are so similar to each other that the confusion that will ensue is going to offset any readability improvement.

I'm less worried about the other name changes for that reason.